### PR TITLE
feat: support bean token holding on combo mainnet

### PR DESCRIPTION
**Pull request by felixfaisal**
This adds new web3network `combo` and supports `holding amount` calculation for bean. 
I couldn't perform e2e test.

### CODE REVIEW

File: tee-worker/litentry/core/common/src/lib.rs [hunk 1]
```diff
@@ -49,5 +49,6 @@ pub fn web3_network_to_chain(network: &Web3Network) -> &'static str {
 		Web3Network::Polygon => "polygon",
 		Web3Network::Arbitrum => "arbitrum",
 		Web3Network::Solana => "solana",
+		Web3Network::Combo => "combo-mainnet",
```
Comment (Kailai-Wang) [inline tee-worker/litentry/core/common/src/lib.rs]: Maybe we should use `combo` here - similar to `bsc`

File: tee-worker/litentry/core/data-providers/src/achainable.rs [hunk 2]
```diff
@@ -188,6 +188,7 @@ pub fn web3_network_to_chain(network: &Web3Network) -> String {
 		Web3Network::Polygon => "polygon".into(),
 		Web3Network::Arbitrum => "arbitrum".into(),
 		Web3Network::Solana => "solana".into(),
+		Web3Network::Combo => "combo-mainnet".into(),
```
Comment (Kailai-Wang) [inline tee-worker/litentry/core/data-providers/src/achainable.rs]: same here - although we might not use it for achainable

File: tee-worker/litentry/core/data-providers/src/nodereal_jsonrpc.rs [hunk 3]
```diff
@@ -91,6 +95,7 @@ impl NoderealChain {
 			NoderealChain::Aptos => "aptos",
 			NoderealChain::Opt => "opt",
 			NoderealChain::Polygon => "polygon",
+			NoderealChain::Combo => "combo-mainnet",
```
Comment (Kailai-Wang) [inline tee-worker/litentry/core/data-providers/src/nodereal_jsonrpc.rs]: It has to be `combo` here - the `mainnet` is appended by default into the url

**Review by Kailai-Wang — CHANGES_REQUESTED** [review]

**Comment by Kailai-Wang** [comment]
Another thing is: we are adding new item to `Web3Network`, I assume this won't require any state migration, right? (meaning the stored user IDGraph could still be retrieved after this upgrade)

**Comment by Kailai-Wang** [comment]
What's still missing is we need to **add** token amount on Bsc and Combo to get the final balance, similar to e.g. LIT - so you need to modify here

```
	match token_type {
		Web3TokenType::Bnb => bnb_balance::get_balance(addresses, data_provider_config),
		Web3TokenType::Eth => eth_balance::get_balance(addresses, data_provider_config),
		Web3TokenType::Lit => lit_balance::get_balance(addresses, data_provider_config),
		Web3TokenType::Sol => sol_balance::get_balance(addresses, data_provider_config),
		Web3TokenType::Btc => btc_balance::get_balance(addresses, data_provider_config),
		_ => common::get_balance(addresses, token_type, data_provider_config),
```

File: tee-worker/litentry/core/common/src/lib.rs (hunk 1 above)
Comment (felixfaisal) [reply tee-worker/litentry/core/common/src/lib.rs]: Updated

File: tee-worker/litentry/core/data-providers/src/achainable.rs (hunk 2 above)
Comment (felixfaisal) [reply tee-worker/litentry/core/data-providers/src/achainable.rs]: Updated

File: tee-worker/litentry/core/data-providers/src/nodereal_jsonrpc.rs (hunk 3 above)
Comment (felixfaisal) [reply tee-worker/litentry/core/data-providers/src/nodereal_jsonrpc.rs]: Updated

**Review by Kailai-Wang — APPROVED** [review]

**Pull request merged**